### ISSUE TITLE
jsdom incorrectly thinks a string is a valid CSS selector

### ISSUE THREAD

**Issue by esr360** [jsdom/jsdom#2285]
### Basic info:

- **Node.js version:** 6.11.4
- **jsdom version:** 11.0.0

### Minimal reproduction case

```js
function isValidSelector(selector) {
    var stub = document.createElement('br');
    
    try { 
        stub.querySelector(selector); 
    } catch(e) { 
        return false; 
    }

    return true;
}

// Returns `false` in a browser, returns `true` in jsdom
console.log(isValidSelector('assets/images/logo-light.png'));
```

### How does similar code behave in browsers?

The above function when passed a value of `assets/images/logo-light.png` will return `false` in browsers, but returns `true` in jsdom.

**Comment by Zirro** [comment]
This comes down to the `shouldAlwaysSelectNothing()` optimisation: https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/nodes/ParentNode-impl.js#L61

Since the `br` element has no children, the fast path is taken before the validity of the selector is tested by nwsapi (which does throw as intended).

Removing the optimisation would be the easiest option, though it does come with (at least) a slight cost to performance. I'll look further into other options later on.

**Comment by esr360** [comment]
Ok thanks! In the mean time can you think of any work arounds?

What I'm currently having to do is import jQuery and change the function to:

```js
function isValidSelector(selector) {
    if (typeof(selector) !== 'string') {
        return false;
    }
    try {
        $(selector);
    } catch(error) {
        return false;
    }
    return true;
}
```

Which seems to work.

**Comment by esr360** [comment]
@Zirro I completely agree that having to include jsdom to determine the validity of a selector seems overkill, however, it is functionality that I require, and like yourself I was unable to determine another method. I think such a thing would definitely be useful as a stand alone package.

**Comment by dperini** [comment]
I am evaluating this request, I might be able to release a standalone package by reducing `nwsapi` to only parse and validate a string, then return true if it passes selectors validation constrains or false if it does not. Thank you for the suggestion.

**Comment by esr360** [comment]
@dperini please do if it's not too much trouble. To reiterate, I think the ability to determine if a string is a valid CSS selector is very useful, and the ability to do this without the need of a DOM would be great.

**Comment by dperini** [comment]
@esr360,
it took a while but as promised I could extract the "isValidSelector()" function you requested.

You can find the package in this GitHub repository: https://github.com/dperini/is-Valid-Selector

Your review of it and testing would be awesome since you have already an environment for this.
As soon as you can confirm this fits your case and requirements I can publish it on `npm`.
Thank you for the suggestion.

**Comment by esr360** [comment]
@dperini thank you so much for taking the time to do this, it is much appreciated. It may take me some time to get around to reviewing and testing but I will be sure to update you with my results.

Thanks again.
